src/clj_sqs_extended/aws/sqs.clj

```diff
@@ -155,29 +169,40 @@
   (->> (DeleteMessageRequest. queue-url (:receiptHandle message))
        (.deleteMessage sqs-client)))
 
+(defn- get-serdes-format-attribute
+  [message]
+  (when message
+    (let [attributes (.getMessageAttributes message)
+          format (get attributes clj-sqs-ext-format-attribute)]
+      (keyword (.getStringValue format)))))
+
+(defn- extract-relevant-keys
+  [message]
+  (if message
+    (-> (bean message)
+        (select-keys [:messageId :receiptHandle :body]))
+    {}))
+
 (defn receive-message
   ([sqs-client queue-url]
    (receive-message sqs-client queue-url {}))
 
   ([sqs-client queue-url
-    {:keys [format
-            wait-time]
-     :or   {format    :transit
-            wait-time 0}}]
-   (letfn [(extract-relevant-keys [message]
-             (if message
-               (-> (bean message)
-                   (select-keys [:messageId :receiptHandle :body]))
-               {}))]
-     (let [request (doto (ReceiveMessageRequest. queue-url)
-                     (.setWaitTimeSeconds (int wait-time))
-                     ;; WATCHOUT: This is a design choice to read one message at a time from the queue
-                     (.setMaxNumberOfMessages (int 1)))
-           response (.receiveMessage sqs-client request)
-           message (->> (.getMessages response) (first) (extract-relevant-keys))]
-       (if-let [payload (serdes/deserialize (:body message) format)]
-         (assoc message :body payload)
-         message)))))
+    {:keys [wait-time]
+     :or   {wait-time 0}}]
+   (let [request (doto (ReceiveMessageRequest. queue-url)
+                   (.setWaitTimeSeconds (int wait-time))
+                   ;; WATCHOUT: Without the next line our custom SerdesFormat attribute will not be received!
+                   (.setAttributeNames ["All"])
+                   ;; WATCHOUT: The next line is a design choice to read one message at a time from the queue
+                   (.setMaxNumberOfMessages (int 1)))
+         response (.receiveMessage sqs-client request)
+         sqs-message (first (.getMessages response))
```

Comment:
is this ok? we're getting messages (plural) but discarding the rest and only using `first`? is there something like `.getOneMessage` or something?

Reply:
We're using `(.setMaxNumberOfMessages (int 1)))` so I would hope the SDK follows and only gives us one message here.

Reply:
oh I missed that. could you leave a comment here please?

Reply:
Sorry, where is "here"? :confused:

Reply:
on L200

Reply:
Ah, sure.